nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/OpenPGPKeyBasedEncryptor.java

```diff
@@ -85,49 +85,61 @@ public StreamCallback getDecryptionCallback() throws Exception {
      */
-    public static boolean validateKeyring(String provider, String secretKeyringFile, char[] passphrase) throws IOException,
+    public static boolean validateKeyring(String provider,
+            String secretKeyringFile, char[] passphrase) throws IOException,
             PGPException, NoSuchProviderException {
-        PGPSecretKeyRingCollection pgpsec = new PGPSecretKeyRingCollection(PGPUtil.getDecoderStream(Files.newInputStream(Paths
-                .get(secretKeyringFile))));
-        Iterator ringit = pgpsec.getKeyRings();
-        while (ringit.hasNext()) {
-            PGPSecretKeyRing secretkeyring = (PGPSecretKeyRing) ringit.next();
-            PGPSecretKey secretkey = secretkeyring.getSecretKey();
-            secretkey.extractPrivateKey(passphrase, provider);
-            return true;
+        try (InputStream fin = Files.newInputStream(Paths
+                .get(secretKeyringFile))) {
+            try (InputStream pin = PGPUtil.getDecoderStream(fin)) {
+                PGPSecretKeyRingCollection pgpsec = new PGPSecretKeyRingCollection(
+                        pin);
+                Iterator ringit = pgpsec.getKeyRings();
+                while (ringit.hasNext()) {
+                    PGPSecretKeyRing secretkeyring = (PGPSecretKeyRing) ringit
+                            .next();
+                    PGPSecretKey secretkey = secretkeyring.getSecretKey();
+                    secretkey.extractPrivateKey(passphrase, provider);
+                    return true;
+                }
+                return false;
+            }
         }
-        return false;
     }
 
     /*
      * Get the public key for a specific user id from a keyring.
      */
     @SuppressWarnings("rawtypes")
-    public static PGPPublicKey getPublicKey(String userId, String publicKeyring) throws IOException, PGPException {
+    public static PGPPublicKey getPublicKey(String userId, String publicKeyring)
+            throws IOException, PGPException {
         PGPPublicKey pubkey = null;
-        PGPPublicKeyRingCollection pgppub = new
-                PGPPublicKeyRingCollection(PGPUtil.getDecoderStream(Files.newInputStream(Paths.get(publicKeyring))));
-
-        Iterator ringit = pgppub.getKeyRings();
-        while (ringit.hasNext()) {
-            PGPPublicKeyRing kring = (PGPPublicKeyRing) ringit.next();
-
-            Iterator keyit = kring.getPublicKeys();
-            while (keyit.hasNext()) {
-                pubkey = (PGPPublicKey) keyit.next();
-                boolean userIdMatch = false;
-
-                Iterator userit = pubkey.getUserIDs();
-                while (userit.hasNext()) {
-                    String id = userit.next().toString();
-                    if (id.contains(userId)) {
-                        userIdMatch = true;
-                        break;
+        try (InputStream fin = Files.newInputStream(Paths.get(publicKeyring))) {
+            try (InputStream pin = PGPUtil.getDecoderStream(fin)) {

```

Comment:
I would wrap this up with the previous try-with-resources so that we have:

try (InputStream fin = Files.newInputStream(Paths.get(publicKeyring)); 
      InputStream pin = PGPUtil.getDecoderStream(fin)) { ...

rather than having a new try-with-resources embedded within the outer one.